specification/dartLangSpec.tex

```diff
@@ -13921,62 +13931,143 @@ \subsection{Exports}
 \IndexCustom{exports a declaration}{exports!declaration})
 if the declaration is in the library's exported namespace.
 
+%% TODO(eernst): Move this to `Imports` when that section gets update: Seems
+%% natural to define it at the first usage in terms of the page number.
 \LMHash{}%
-A library always exports all names and all declarations in its public namespace.
-In addition, a library may choose to re-export additional libraries via \Index{export directives}, often referred to simply as \Index{exports}.
+We need to combine two namespaces,
+which is an operation that is defined
+when they have disjoint sets of keys:
+The
+\IndexCustom{union of two namespaces}{namespace!union},
+\IndexCustom{$\Namespace{a}\cup\Namespace{b}$}{%
+  $\cup$@$\Namespace{a} \cup \Namespace{b}$},
+is the namespace that maps
+each key $n$ of \Namespace{a}
+to the corresponding value $\Namespace{a}(n)$,
+and each key $n$ of \Namespace{b}
+to $\Namespace{b}(n)$.
 
 \LMHash{}%
-Let $E$ be an export directive that refers to a URI via the string $s_1$.
-Evaluation of $E$ proceeds as follows:
+The exported namespace of a library is then determined as follows.
+Let $L$ be a library,
+let \List{E}{1}{m} be the export directives of $L$,
+and let \List{L}{1}{m} be libraries
+such that $E_i$ refers to $L_i$ for all $i \in 1 .. m$.
 
 \LMHash{}%
-First,
+Let \Namespace{L} be the namespace that
+for each public top-level declaration $D$ with name $n$ in $L$ maps $n$ to $D$.
+For each $i \in 1 .. m$,
+let $\Namespace{\metavar{export}, i}$ be the exported namespace of $L_i$,
+and let \Namespace{i} be a namespace obtained from
+$\Namespace{\metavar{export}, i}$ by the narrowing process described below.
 
-\begin{itemize}
-\item
-If the URI that is the value of $s_1$ has not yet been accessed by an import or export directive in the current isolate then the contents of the URI are compiled to yield a library $B$.
-\item Otherwise, the contents of the URI denoted by $s_1$ have been compiled into a library $B$ within the current isolate.
-\end{itemize}
+\LMHash{}%
+The narrowing process that yields
+\Namespace{i} from $\Namespace{\metavar{export}, i}$
+consists of two steps,
+each of them restricting the domain of the namespace to a subset
+(\commentary{%
+that is, deleting the binding from a key to a value,
```

Comment:
"that is, deletes"

Reply:
Adjusted both `restricting` and `deleting`.